slack-plugin/src/main/kotlin/slack/gradle/lint/LintTasks.kt

```diff
@@ -87,139 +255,196 @@ internal object LintTasks {
         null
       }
 
-    // We only want to create tasks once, but a project might apply multiple plugins.
-    val applied = AtomicBoolean(false)
+    globalTask?.configure { dependsOn(ciLintTask) }
 
-    if (commonExtension != null) {
-      project.log("Applying ciLint to Android project")
-      applyCommonLints()
-      createAndroidCiLintTask(
-        project,
-        globalTask,
-        slackProperties,
-        commonExtension,
-        sdkVersions!!.invoke()
-      )
-    } else {
-      val javaKotlinLibraryHandler = { _: AppliedPlugin ->
-        if (applied.compareAndSet(false, true)) {
-          // Enable linting on pure JVM projects
-          project.pluginManager.apply("com.android.lint")
-          project.configure<Lint> { configureLint(project, slackProperties, null, false) }
-          applyCommonLints()
-          project.log("Creating ciLint task")
-          val ciLint =
-            project.tasks.register(COMPILE_CI_LINT_NAME) {
-              group = LifecycleBasePlugin.VERIFICATION_GROUP
-              dependsOn("lint")
-            }
-          globalTask?.configure { dependsOn(ciLint) }
-        }
-      }
-      project.pluginManager.withPlugin("java-library", javaKotlinLibraryHandler)
-      project.pluginManager.withPlugin("org.jetbrains.kotlin.jvm", javaKotlinLibraryHandler)
-    }
-  }
+    afterEvaluate { addSourceSetsForAndroidMultiplatformAfterEvaluate() }
 
-  private fun createAndroidCiLintTask(
-    project: Project,
-    globalTask: TaskProvider<*>?,
-    slackProperties: SlackProperties,
-    extension: CommonExtension<*, *, *, *, *>,
-    sdkVersions: SlackProperties.AndroidSdkProperties,
-  ) {
-    project.log("Configuring android lint tasks. isApp=${extension is AppExtension}")
-    extension.lint {
-      configureLint(
-        project,
-        slackProperties,
-        sdkVersions,
-        checkDependencies = extension is AppExtension
-      )
+    lint.apply {
+      // Skip lintVital tasks on assemble. We explicitly run lintRelease for libraries.
+      checkReleaseBuilds = false
```

Comment:
is there a reason to use `apply {}` here when you're just configuring one property? Or should you collapse this block with the block below?

Reply:
[2d95b59](https://github.com/slackhq/slack-gradle-plugin/pull/612/commits/2d95b592410f3722cf4099a6fa1377a1e114cf6e)